src/cascadia/TerminalApp/MinMaxCloseControl.xaml

```diff
@@ -10,12 +10,283 @@
+                                                </ObjectAnimationUsingKeyFrames>
+                                            </Storyboard>
+                                        </VisualState>
+
+                                        <VisualState x:Name="Disabled">
+
+                                            <Storyboard>
+                                                <ObjectAnimationUsingKeyFrames Storyboard.TargetName="ButtonBaseElement" Storyboard.TargetProperty="Background">
+                                                    <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ButtonBackgroundDisabled}" />
+                                                </ObjectAnimationUsingKeyFrames>
+                                                <ObjectAnimationUsingKeyFrames Storyboard.TargetName="ButtonBaseElement" Storyboard.TargetProperty="BorderBrush">
+                                                    <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ButtonBorderBrushDisabled}" />
+                                                </ObjectAnimationUsingKeyFrames>
+                                            </Storyboard>
+                                        </VisualState>
+
+                                    </VisualStateGroup>
+                                    <VisualStateGroup x:Name="MinMaxStates">
+                                        <VisualState x:Name="WindowStateNormal" />
+
+                                        <VisualState x:Name="WindowStateMaximized">
+                                            <VisualState.Setters>
+                                                <Setter Target="Path.Data" Value="M 0 2 h 8 v 8 h -8 v -8 M 2 2 v -2 h 8 v 8 h -2" />
+                                            </VisualState.Setters>
+                                        </VisualState>
+                                    </VisualStateGroup>
+                                </VisualStateManager.VisualStateGroups>
+
+                                <Path
+                                    x:Name="Path"
+                                    StrokeThickness="1"
+                                    Stroke="{ThemeResource CaptionButtonStrokeBrush}"
+                                    Data="M 0 0 h 10 v 10 h -10 v -10"
+                                    Stretch="Fill"
+                                    UseLayoutRounding="False"
+                                    Width="11"
+                                    Height="11"
+                                    StrokeEndLineCap="Square"
+                                    StrokeStartLineCap="Square" />
+                            </Border>
+
+                        </ControlTemplate>
+                    </Setter.Value>
+                </Setter>
+            </Style>
+
+            <!-- CloseButton is a CaptionButton that turns red on hover. We *once again* need to override all
+                 of its VisualStates to make this happen. -->

```

Comment:
Because you need to change the resource references that the VSM states use? How about just using a ResourceDictionary that you set as the Resources property which overrides all the Button Brush resource keys?

Reply:
Well! I just learned that that was possible. Thanks!

Reply:
I think I finally understand how this is supposed to work. That cut down my xaml burden quite a lot, but I'm not 100% sure I did the "path data" thing right. I made it a resource that could be customized in the control's resources, since I couldn't find a way to add a _property_ that I could rely on to stuff the path in.

Reply:
For the path data my recommendation would be to add a custom "CaptionButton" type (derived from Button) that has the Path property which you then bind into the Template with TemplateBinding. And if you really don't want to make a custom UI type you could DependencyProperty.Register and then use your attached property in the style setter.